framework/include/parser/CommandLine.h

```diff
@@ -144,11 +141,17 @@ class CommandLine
 
 protected:
   /**
-   * Used to set the argument value, allows specialization
+   * Helper for setting the argument value, allows specialization
    */
   template <typename T>
   void setArgument(std::stringstream & stream, T & argument);
 
+  /**
+   * Helper for setting the argument value; catches errors so we can provide more context
```

Comment:
I know that `cli_switch` has been in this file for over a decade, so it's not a name you made ... but it would help me to have more doxygen about what `cli_switch` means